src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
Hmm, I think returning just the closingComment was convenient because the `lastPlacedNode` stuff in ReactMultiChild uses `getNativeNode` for that, and I didn't have to add any special cases for it that way. But I'll see if that turns out to be simpler.